gcp_airflow_foundations/base_class/template_ingestion_options_config.py

```diff
@@ -0,0 +1,23 @@
+from dacite import Config
+from pydantic import validator, root_validator
+from pydantic.dataclasses import dataclass
+
+from typing import List, Optional
+import regex as re
+
+from gcp_airflow_foundations.enums.schema_source_type import SchemaSourceType
+
+
+@dataclass
+class TemplateIngestionOptionsConfig:
+    """
+    Attributes:
+        ingest_all_tables: if true, ingest all tables from source
+        ingestion_name: name to use for DAG id, should correspond to the relevant partition of the source cut out by the regex expression provided
+        dag_creation_mode: if "SOURCE", then one DAG per source is created. if "TABLE", one DAG per table.
+        regex_table_matching: regex pattern to match tables to, if ingest_all_tables is false
+    """
+    ingest_all_tables: bool = False
+    ingestion_name: str = ""
```

Comment:
why is default ""?

Reply:
Honestly, probably shouldn't even give it a default - there is no "default" ingestion name. Was just testing on best way to get it to work - default values are provided in the SourceTemplateConfig anyway, so it won't break

Reply:
Can make it optional

Reply:
So what's the conclusion on this? @danielkulikov

Reply:
updated this to just be a mandatory field w/ no default, it should be provided